frontend/src/components/common/Resource/ResourceTable.tsx

```diff
@@ -114,7 +114,7 @@ export default function ResourceTable<RowItem>(
 
 function TableFromResourceClass<RowItem>(props: ResourceTableFromResourceClassProps<RowItem>) {
   const { resourceClass, id, ...otherProps } = props;
-  const [items, error] = resourceClass.useList();
+  const { items, error } = resourceClass.useListQuery();
```

Comment:
That would be a very nice improvement. Query also returns a `status` string which I think is even better than `isLoading`. But currently in the code we rely on null checks a lot and it would require a big refactor which I think should be done separately

Reply:
Status is fine too. Now that we return an object, it's fine to have these added later.

Reply:
BTW, if we are returning anything that are implementation details from react-query? (Maybe that `status`) I think we should probably not do that ATM and just stick with some basics till we figure out what makes sense to return, unless you tell me we are limiting a lot the possibilities because of that.

Reply:
added an interface, so we don't have to rely on react-query